src/remote-node/client.js

```diff
@@ -0,0 +1,317 @@
+    this._id = id
+    this._apiAddr = multiaddr(apiAddr)
+    this._gwAddr = multiaddr(gwAddrs)
+    this.initialized = false
+    this.started = false
+    this.api = createApi(apiAddr, gwAddrs)
+  }
+
+  /**
+   * Get the address of connected IPFS API.
+   *
+   * @returns {Multiaddr}
+   */
+  get apiAddr () {
+    return this._apiAddr
+  }
+
+  /**
+   * Set the address of connected IPFS API.
+   *
+   * @param {Multiaddr} addr
+   * @returns {void}
+   */
+  set apiAddr (addr) {
+    this._apiAddr = addr
+  }
+
+  /**
+   * Get the address of connected IPFS HTTP Gateway.
+   *
+   * @returns {Multiaddr}
+   */
+  get gatewayAddr () {
+    return this._gwAddr
+  }
+
+  /**
+   * Set the address of connected IPFS Gateway.
+   *
+   * @param {Multiaddr} addr
+   * @returns {void}
+   */
+  set gatewayAddr (addr) {
+    this._gwAddr = addr
+  }
+
+  /**
+   * Initialize a repo.
+   *
+   * @param {Object} [initOpts={}]
+   * @param {number} [initOpts.keysize=2048] - The bit size of the identiy key.
+   * @param {string} [initOpts.directory=IPFS_PATH] - The location of the repo.
+   * @param {function (Error, Node)} cb
+   * @returns {undefined}
+   */
+  init (initOpts, cb) {
+    if (typeof initOpts === 'function') {
+      cb = initOpts
+      initOpts = {}
+    }
+
+    request
+      .post(`${this.baseUrl}/init`)
+      .query({ id: this._id })
+      .send({ initOpts })
+      .end((err, res) => {
+        if (err) {
+          return cb(new Error(err.response ? err.response.body.message : err))
+        }
+
+        this.initialized = res.body.initialized
+        cb(null, res.body)
+      })
+  }
+
+  /**
+   * Delete the repo that was being used.
+   * If the node was marked as `disposable` this will be called
+   * automatically when the process is exited.
+   *
+   * @param {function(Error)} cb
+   * @returns {undefined}
+   */
+  cleanup (cb) {
+    request
+      .post(`${this.baseUrl}/cleanup`)
+      .query({ id: this._id })
+      .end((err) => { cb(err) })
+  }
+
+  /**
+   * Start the daemon.
+   *
+   * @param {Array<string>} [flags=[]] - Flags to be passed to the `ipfs daemon` command.
+   * @param {function(Error, IpfsApi)} cb
+   * @returns {undefined}
+   */
+  start (flags, cb) {
+    if (typeof flags === 'function') {
+      cb = flags
+      flags = []
+    }
+
+    request
+      .post(`${this.baseUrl}/start`)
+      .query({ id: this._id })
+      .send({ flags })
+      .end((err, res) => {
+        if (err) {
+          return cb(new Error(err.response ? err.response.body.message : err))
+        }
+
+        this.started = true
+
+        const apiAddr = res.body.api ? res.body.api.apiAddr : ''
+        const gatewayAddr = res.body.api ? res.body.api.gatewayAddr : ''
+
+        this.api = createApi(apiAddr, gatewayAddr)
+        return cb(null, this.api)
+      })
+  }
+
+  /**
+   * Stop the daemon.
+   *
+   * @param {function(Error)} cb
+   * @returns {undefined}
+   */
+  stop (cb) {
```

Comment:
On the daemon the callback is optional. Not sure if it should also be optional on the client.

Reply:
good catch, fixed.